crates/driver/src/domain/competition/solution/mod.rs

```diff
@@ -96,14 +98,27 @@ impl Solution {
                             buy: solution.prices
                                 [&fulfillment.order().buy.token.wrap(solution.weth)],
                         };
-                        let fulfillment = fulfillment.with_protocol_fees(prices)?;
-                        trades.push(Trade::Fulfillment(fulfillment))
+                        let trade = trade.with_protocol_fees(prices)?;
+                        trades.push(trade)
                     }
                     order::Kind::Liquidity => {
                         trades.push(trade);
                     }
                 },
-                Trade::Jit(_) => trades.push(trade),
+                Trade::Jit(jit) => {
+                    if cow_amm_addresses.contains(&jit.order().receiver.into()) {
```

Comment:
The important part to check is the orders `signer`. I assume the `receiver` of the CoW AMM orders have to be `0x00000` to avoid creating orders which can send the proceeds of a trade outside of the contract.